app/src/main/java/org/fossasia/pslab/fragment/LuxMeterFragmentData.java

```diff
@@ -0,0 +1,222 @@
+import android.hardware.SensorEventListener;
+import android.hardware.SensorManager;
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.EditText;
+
+import com.github.anastr.speedviewlib.PointerSpeedometer;
+import com.github.mikephil.charting.charts.LineChart;
+import com.github.mikephil.charting.components.Legend;
+import com.github.mikephil.charting.components.XAxis;
+import com.github.mikephil.charting.components.YAxis;
+import com.github.mikephil.charting.data.Entry;
+import com.github.mikephil.charting.data.LineData;
+import com.github.mikephil.charting.data.LineDataSet;
+
+import org.fossasia.pslab.PSLabApplication;
+import org.fossasia.pslab.R;
+import org.fossasia.pslab.communication.sensors.BH1750;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import static android.content.Context.SENSOR_SERVICE;
+
+public class LuxMeterFragmentData extends Fragment {
+
+    private SensorDataFetch sensorDataFetch;
+    private EditText statMax;
+    private EditText statMin;
+    private EditText statMean;
+    private BH1750 sensorBH1750 = null;
+    private SensorManager sensorManager;
+    private Sensor sensor;
+    private LineChart mChart;
+    private long startTime;
+    private int flag;
+    private ArrayList<Entry> entries;
+    private Runnable runnable;
+    private Thread dataThread;
+    private PointerSpeedometer lightMeter;
+    private float currentMin;
+    private float currentMax;
+    private YAxis y;
+
+
+    public static LuxMeterFragmentData newInstance() {
+        return new LuxMeterFragmentData();
+    }
+
+    public LuxMeterFragmentData() {
+        // Required empty public constructor
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setRetainInstance(true);
+        currentMin = 2000;
+        entries = new ArrayList<>();
+        sensorManager = (SensorManager) getContext().getSystemService(SENSOR_SERVICE);
+        sensor = sensorManager.getDefaultSensor(Sensor.TYPE_LIGHT);
+
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_lux_meter_data, container, false);
+
+        runnable = new Runnable() {
+            @Override
+            public void run() {
+                sensorDataFetch = new SensorDataFetch();
+                sensorDataFetch.execute();
+                if (flag == 0) {
+                    startTime = System.currentTimeMillis();
+                    flag = 1;
+                }
+            }
+        };
+        dataThread = new Thread(runnable);
+        dataThread.run();
+
+        statMax = (EditText) view.findViewById(R.id.lux_stat_max);
```

Comment:
These values can't seem to be changed. So a Textview would be better rather than a Edittext